sympy/solvers/solveset.py

```diff
@@ -432,7 +436,17 @@ def _solve_as_poly(f, symbol, domain=S.Complexes):
                 else:
                     result = ConditionSet(symbol, Eq(f, 0), domain)
         else:
-            result = ConditionSet(symbol, Eq(f, 0), domain)
+            if all(_ispow(g) for g in gens):
```

Comment:
Thanks for your time for the PR.
The proper way to solve this  `5**(x**2) = 5**(6-x)` is taking log both side then we get `5**(x**2 -6+x) =0` , here I tried : 
```
>>> powsimp((5**(x**2) - 5**(6-x)).factor(5**(x**2)))
    ⎛  2            ⎞
 -x ⎜ x  + x        ⎟
5  ⋅⎝5       - 15625⎠
>>> q =powsimp((5**(x**2) - 5**(6-x)).factor(5**(x**2)))
>>> invert_real(q,0,x)
⎛    ⎛  2            ⎞     ⎞
⎜ -x ⎜ x  + x        ⎟     ⎟
⎝5  ⋅⎝5       - 15625⎠, {0}⎠
>>> solveset_real(q-0,x)
{-3, 2}
```

But now as I am trying to implement `lambertW` in solveset. I think this line of code should be during checking of eq for `LambertW` process or may be this will be fixed automatically.
For `LambertW` , I am trying to use `_solve_lambert` that is defined inside the `bivariate.py` as the `solver` do. 
If you have suggestions regarding `LambertW` ,please share. @hargup